Added referrals invite notification

aileen opened the pull request:
no issue

- Ghost users that make >= $100 MRR will see a dismissible notification that invites them to the Ghost Referral program
- Only Admin and Owner users will see it
- Only applies when Stripe is setup and connected in live mode
- By saving a `referralInviteDismissed` property to the users' `accessibility` JSON object we can determine if the notification has been dismissed and won't show it again
- Added new `gh-referral-invite` component

aileen committed:
Show referrals program invite for sites > $1k MRR

closes https://www.notion.so/ghost/Referral-program-invite-in-admin-8c6953e7312a41e49ba5565143ace80e

- Only show when no theme errors are present, as it’s using the same markup for the box
- Open referrals link in a new tab
- Only show when MRR is > $1,000 and Stripe is connected in live mode

aileen committed:
Simplify referrals notification by turning it into an anchor link

aileen committed:
Check MRR is > $100

aileen committed:
Updated styles and markup

aileen committed:
Remove transition on close button for referral toast

aileen committed:
Added ability to dismiss referral invite notification

- By saving a `referralInviteDismissed` property to the users `accessibility` JSON object we can determine if the notification has been dismissed and won't show it again
- Added condition to only show the notification to Admins and Owners

aileen committed:
Ensure current MRR is loaded without

kevinansfield commented on ghost/admin/app/components/gh-nav-menu/footer.js:
```diff
@@ -35,6 +63,22 @@ export default class Footer extends Component {
+    dismissReferralInvite(event) {
+        event.preventDefault();
+        event.stopPropagation();
+        const key = 'referralInviteDismissed';
+        const value = moment().tz(this.settings.timezone);
+        const options = {user: true};
+
+        return this.feature.update(key, value, options);
```
`this.feature.update` is more of an internal implementation detail and shouldn't be used outside of the feature service.

Inside the feature service you can add

```js
@feature('referralInviteDismissed', {user: true}) referralInviteDismissed;
```

then in this code you can use that directly, e.g.

```js
if (!this.feature.referralInviteDismissed) {}`

this.feature.referralInviteDismissed = moment().tz(this.settings.timezone)
```

kevinansfield commented on ghost/admin/app/components/gh-nav-menu/footer.hbs:
```diff
@@ -1,9 +1,16 @@
+        <span class="gh-footer-toast-title gh-notification-title">Your theme contains errors</span>
+        <span class="gh-footer-toast-p">Some functionality on your site may be limited &rarr;</span>
     </button>
+    {{else if (and this.showReferralInvite (not this.hasThemeErrors))}}
```
This can be simplified, the `not this.hasThemeErrors` is already handled by the `if/else`

kevinansfield commented on ghost/admin/app/components/gh-nav-menu/footer.hbs:
```diff
@@ -1,9 +1,16 @@
+        <span class="gh-footer-toast-p">Some functionality on your site may be limited &rarr;</span>
     </button>
+    {{else if (and this.showReferralInvite (not this.hasThemeErrors))}}
+    <a href="https://referrals.ghost.org/?ref=admin" class="gh-referral-toast" target="_blank" rel="noopener noreferrer">
+        <span class="gh-referral-toast-close" {{on "click" (fn this.dismissReferralInvite)}}>&#xd7;</span>
+        <strong>You qualify for our invite-only  referral program.</strong>
+        <p class="gh-footer-toast-p">Refer people to Ghost and earn a <strong>30% share</strong> of the subscription revenue, every single month.</p>
+        <button class="gh-btn gh-btn-black gh-referral-toast-button"><span>Find out more &rarr;</span></button>
+    </a>
```
It would be cleaner to pull this into a separate component, then you can write it as a glimmer component and use it's `constructor` to load the MRR stats rather than `{{did-insert}}` which we generally want to avoid (it's designed as a band-aid whilst migrating from old-style to new-style Ember and is [generally considered an anti-pattern](https://github.com/emberjs/ember-render-modifiers#when-to-use-these-modifiers-and-when-not-to-use-them)).

Means there's less old-style code that needs migrating later down the line and the show/not-show/dismiss logic is all self-contained rather than being mixed in with the other nav footer code 😄 (+ easier to test, easier to move when designs change)

aileen committed:
Added method to feature service to update `referralInviteDismissed`

refs https://github.com/TryGhost/Ghost/pull/16187#discussion_r1087251797

aileen committed:
Moved `hasThemeErrors` conditional to computed prop

refs https://github.com/TryGhost/Ghost/pull/16187#discussion_r1087253904

aileen committed:
Moved referral invite into glimmer component

refs https://github.com/TryGhost/Ghost/pull/16187#discussion_r1087258005

- Moved the relevant pieces for the Referral notification into a new `gh-referral-invite` component

aileen commented on ghost/admin/app/components/gh-nav-menu/footer.js:
```diff
@@ -11,8 +11,6 @@ import {inject as service} from '@ember/service';
     @service session;
     @service router;
-    @service whatsNew;
-    @service feature;
```
There were not used in the footer component

aileen committed:
Fixed referral toast close button in dark mode

aileen committed:
Fixed linting errors caused my markup

aileen committed:
Fixed tests
